src/php/Printer/TypePrinter/ArrayPrinter.php

```diff
@@ -23,20 +25,15 @@ public function __construct(PrinterInterface $printer, bool $withColor = false)
     /**
      * @param array $form
      */
-    public function print($form): string
+    public function print(mixed $form): string
     {
-        $arr = $this->isList($form)
+        $arr = array_is_list($form)
```

Comment:
This function is only available in PHP 8.1 not PHP 8.0. See: https://www.php.net/manual/de/function.array-is-list.php

There is probably no tests for this?

Reply:
Wow, good catch! Actually we do have tests for this, see `ArrayPrinterTest`. After some investigation, I found why this is working: because we have this in vendor
```php
# source: vendor/symfony/polyfill-php81/bootstrap.php
if (!function_exists('array_is_list')) {
    function array_is_list(array $array): bool { return p\Php81::array_is_list($array); }
}
```
And `Php81::array_is_list` has the same behaviour as the native function, but written in PHP:
```php
<?php
# source: vendor/symfony/polyfill-php81/Php81.php
namespace Symfony\Polyfill\Php81;

final class Php81
{
    public static function array_is_list(array $array): bool
    {
        if ([] === $array || $array === array_values($array)) {
            return true;
        }

        $nextKey = -1;

        foreach ($array as $k => $v) {
            if ($k !== ++$nextKey) {
                return false;
            }
        }

        return true;
    }
}
```

And why do we have that code in the vendor? because it's requires from `friendsofphp/php-cs-fixer  v3.7.0`:
```bash
phel-lang/ (refactor/printer-module-using-php80) $ composer why symfony/polyfill-php81
friendsofphp/php-cs-fixer  v3.7.0  requires  symfony/polyfill-php81 (^1.25)
```
---

I will simply rollback in order to have our custom `array_is_list` as we had before, in order to keep as less external dependencies as necessary.